File: app/src/main/java/com/pedrosoares/cotacaoapp/core/base/BasePresenter.java
package com.pedrosoares.cotacaoapp.core.base;

import io.reactivex.disposables.CompositeDisposable;

public abstract class BasePresenter implements BaseContract.Presenter {

    protected CompositeDisposable compositeDisposable;

    @Override
    public void onAttach() {
        compositeDisposable = new CompositeDisposable();
    }

    @Override
    public void onDetach() {
        if (compositeDisposable != null) {
            compositeDisposable.clear();
        }
    }
}

File: app/src/main/java/com/pedrosoares/cotacaoapp/core/base/BasePresenter.kt
package com.pedrosoares.cotacaoapp.core.base

import com.pedrosoares.cotacaoapp.core.base.BaseContract.Presenter
import io.reactivex.disposables.CompositeDisposable

abstract class BasePresenter : Presenter {
    protected var compositeDisposable: CompositeDisposable? = null

    override fun onAttach() {
        compositeDisposable = CompositeDisposable()
    }

    override fun onDetach() {
        compositeDisposable?.let { compositeDisposable!!.clear()   }
    }
}